pkg/payloads/upload.go

```diff
@@ -12,3 +14,9 @@ type Module struct {
 	Name    string `json:"name"`
 	Version string `json:"version"`
 }
+
+// PushNotification is used to notify other Olympus instances about a new event
+type PushNotification struct {
+	Events    []eventlog.Event `json:"events"`
```

Comment:
i'm in doubts here, push notification should be immediate action on a processed cache miss. it should send module, version and CDN uri where the package is.
and at this point you don't know subset of eventlog receiver is interested in as receiver holds the pointer of last acquired entry
so i guess, send single event. process as regular cachemiss but do not send push notification out.
db merge and subset of eventlog will be retrieved on regular pulls
@arschles

Reply:
I assumed the events slice will contain single element but I thought using slice leaves the possibility to maybe send those in batches.

I'm not sure I understood it right but if we process the push event as a cache miss wouldn't we download the module using the generic/git fetcher? I thought I'll use mergedb because it already uses the CDN Uri.

Reply:
yes yes, of course, we need to use redirect uri from the request. by processing as regular cachemiss i meant, that you have a worker processing single events holding (module, version, uri). not well-picked comparison, my bad.
so to summarize, everything in this PR is right, the only thing differing from my state of mind is batch/single instance.

Reply:
Ok, now I get it. My understanding was also that the push notification will be sent after processed cache miss. My consideration was that a slice will be more flexible but I agree that it can be confusing.

Reply:
That is the right understanding @michalpristas @marpio (see https://github.com/gomods/athens/wiki/The-Central-Package-Registry-(Olympus)#push-sync). I agree that a slice is easy to do and more flexible, so I agree with keeping it @marpio